test/libsinsp_e2e/event_capture.h

```diff
@@ -158,7 +153,9 @@ class event_capture {
 
 		if(!capturing.m_start_failed.load()) {
 			run_function(capturing.get_inspector_handle());
-			capturing.stop_capture();
+			// This lseek is just used to be sure that all the events
+			// created in the run_function were consumed.
+			lseek(-1,-1,-1);
```

Comment:
I'd use something less error prone (ie: perhaps there can be something else in the system that calls `lseek(-1`):
* open a "lock file" like `m_fd = open("/tmp/__test_running")`
* at the end, `close(m_fd)`
* wait for the close syscall event on m_fd for our TID

WDYT? It seems easier to understand and safer.